From 3ff0ba16566c029c1f5cf4d6bf5584e95b0661bb Mon Sep 17 00:00:00 2001
From: Todd Short <tshort@akamai.com>
Date: Tue, 5 Jan 2021 13:50:21 -0500
Subject: [PATCH 27/63] QUIC: return success when no post-handshake data

---
 ssl/ssl_quic.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ssl/ssl_quic.c b/ssl/ssl_quic.c
index 971af0f4c5..477a386da3 100644
--- a/ssl/ssl_quic.c
+++ b/ssl/ssl_quic.c
@@ -309,6 +309,10 @@ int SSL_process_quic_post_handshake(SSL *ssl)
         return 0;
     }
 
+    /* if there is no data, return success as BoringSSL */
+    if (ssl->quic_input_data_head == NULL)
+        return 1;
+    
     /*
      * This is always safe (we are sure to be at a record boundary) because
      * SSL_read()/SSL_write() are never used for QUIC connections -- the
-- 
2.42.0

