From 117b887a7a132b25425c74613fc9621dd660fecb Mon Sep 17 00:00:00 2001
From: Todd Short <tshort@akamai.com>
Date: Thu, 24 Jun 2021 15:11:24 -0400
Subject: [PATCH 42/63] QUIC: Fix make doc-nits

---
 doc/man1/openssl-info.pod.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/doc/man1/openssl-info.pod.in b/doc/man1/openssl-info.pod.in
index 0e91bb28ee..fe7abee486 100644
--- a/doc/man1/openssl-info.pod.in
+++ b/doc/man1/openssl-info.pod.in
@@ -17,6 +17,7 @@ B<openssl info>
 [B<-listsep>]
 [B<-seeds>]
 [B<-cpusettings>]
+[B<-quic>]
 
 =head1 DESCRIPTION
 
@@ -73,6 +74,10 @@ Outputs the randomness seed sources.
 
 Outputs the OpenSSL CPU settings info.
 
+=item B<-quic>
+
+Outputs the OpenSSL QUIC info.
+
 =back
 
 =head1 HISTORY
-- 
2.42.0

