--- crypto/bn/bn_lcl.h	2012-01-18 02:40:36.000000000 +0100
+++ crypto/bn/bn_lcl.h.oden	2012-03-14 17:36:07.034007529 +0100
@@ -293,6 +293,15 @@ extern "C" {
 	     : "=l"(low),"=h"(high)	\
 	     : "r"(a), "r"(b));
 #  endif
+# elif defined(__ia64) && defined(SIXTY_FOUR_BIT_LONG)
+#  if defined(__GNUC__)
+#   define BN_UMULT_HIGH(a,b) ({      \
+      register BN_ULONG ret;          \
+      asm ("xmpy.hu %0 = %1, %2"      \
+           : "=f"(ret)                \
+           : "f"(a), "f"(b));         \
+      ret;                    })
+#  endif      /* compiler */
 # endif		/* cpu */
 #endif		/* OPENSSL_NO_ASM */
 
