--- openssl-1.0.0i.orig/apps/openssl.cnf	2011-12-06 04:01:00.000000000 +0400
+++ openssl-1.0.0i/apps/openssl.cnf	2012-04-20 00:19:13.048062134 +0400
@@ -103,7 +103,8 @@
 
 ####################################################################
 [ req ]
-default_bits		= 1024
+default_bits		= 2048
+default_md		= sha1
 default_keyfile 	= privkey.pem
 distinguished_name	= req_distinguished_name
 attributes		= req_attributes
@@ -126,17 +127,18 @@
 
 [ req_distinguished_name ]
 countryName			= Country Name (2 letter code)
-countryName_default		= AU
+countryName_default		= XX
 countryName_min			= 2
 countryName_max			= 2
 
 stateOrProvinceName		= State or Province Name (full name)
-stateOrProvinceName_default	= Some-State
+stateOrProvinceName_default	= Default Province
 
 localityName			= Locality Name (eg, city)
+localityName_default		= Default City
 
 0.organizationName		= Organization Name (eg, company)
-0.organizationName_default	= Internet Widgits Pty Ltd
+0.organizationName_default	= Default Company Ltd
 
 # we can do this but it is not needed normally :-)
 #1.organizationName		= Second Organization Name (eg, company)
