--- openssl-1.1.0f/Configurations/10-main.conf.0001~	2017-05-25 14:46:17.000000000 +0200
+++ openssl-1.1.0f/Configurations/10-main.conf	2017-06-26 11:52:52.697019949 +0200
@@ -662,6 +662,7 @@ sub vms_info {
         cflags           => add("-m64 -DL_ENDIAN"),
         perlasm_scheme   => "linux64le",
         shared_ldflag    => add("-m64"),
+        multilib         => "64",
     },
 
     "linux-armv4" => {
@@ -702,6 +703,7 @@ sub vms_info {
     "linux-aarch64" => {
         inherit_from     => [ "linux-generic64", asm("aarch64_asm") ],
         perlasm_scheme   => "linux64",
+        multilib         => "64",
     },
     "linux-arm64ilp32" => {  # https://wiki.linaro.org/Platform/arm64-ilp32
         inherit_from     => [ "linux-generic32", asm("aarch64_asm") ],
--- openssl-1.1.0f/crypto/ec/ec_lib.c.0001~	2017-05-25 14:46:18.000000000 +0200
+++ openssl-1.1.0f/crypto/ec/ec_lib.c	2017-06-26 11:52:52.697019949 +0200
@@ -74,9 +74,6 @@ void EC_pre_comp_free(EC_GROUP *group)
         break;
 #endif
 #ifndef OPENSSL_NO_EC_NISTP_64_GCC_128
-    case PCT_nistp224:
-        EC_nistp224_pre_comp_free(group->pre_comp.nistp224);
-        break;
     case PCT_nistp256:
         EC_nistp256_pre_comp_free(group->pre_comp.nistp256);
         break;
@@ -152,9 +149,6 @@ int EC_GROUP_copy(EC_GROUP *dest, const
         break;
 #endif
 #ifndef OPENSSL_NO_EC_NISTP_64_GCC_128
-    case PCT_nistp224:
-        dest->pre_comp.nistp224 = EC_nistp224_pre_comp_dup(src->pre_comp.nistp224);
-        break;
     case PCT_nistp256:
         dest->pre_comp.nistp256 = EC_nistp256_pre_comp_dup(src->pre_comp.nistp256);
         break;
--- openssl-1.1.0f/test/ecdsatest.c.0001~	2017-05-25 14:46:21.000000000 +0200
+++ openssl-1.1.0f/test/ecdsatest.c	2017-06-26 11:52:52.697019949 +0200
@@ -216,6 +216,7 @@ int x9_62_tests(BIO *out)
     if (!change_rand())
         goto x962_err;
 
+#if 0
     if (!x9_62_test_internal(out, NID_X9_62_prime192v1,
                              "3342403536405981729393488334694600415596881826869351677613",
                              "5735822328888155254683894997897571951568553642892029982342"))
@@ -226,6 +227,7 @@ int x9_62_tests(BIO *out)
                              "3238135532097973577080787768312505059318910517550078427819"
                              "78505179448783"))
         goto x962_err;
+#endif
 # ifndef OPENSSL_NO_EC2M
     if (!x9_62_test_internal(out, NID_X9_62_c2tnb191v1,
                              "87194383164871543355722284926904419997237591535066528048",
