diff -up openssl-3.0.0-alpha14/Configure.2~ openssl-3.0.0-alpha14/Configure
--- openssl-3.0.0-alpha14/Configure.2~	2021-04-08 14:15:48.000000000 +0200
+++ openssl-3.0.0-alpha14/Configure	2021-04-10 01:07:31.969642338 +0200
@@ -1577,7 +1577,7 @@ if (!$disabled{asm} && !$predefined_C{__
 $config{use_int128} = 0;
 {
     my $cc = $config{CROSS_COMPILE}.$config{CC};
-    open(PIPE, "$cc -E -dM - </dev/null 2>&1 |");
+    open(PIPE, "$cc $target{cflags} -E -dM - </dev/null 2>&1 |");
     while(<PIPE>) {
         if (m/__SIZEOF_INT128__/) {
             $config{use_int128} = 1;
