diff -up openssl-1.1.1/include/openssl/opensslv.h.14~ openssl-1.1.1/include/openssl/opensslv.h
--- openssl-1.1.1/include/openssl/opensslv.h.14~	2018-09-11 21:45:10.006850429 +0200
+++ openssl-1.1.1/include/openssl/opensslv.h	2018-09-11 21:46:29.291906386 +0200
@@ -40,7 +40,7 @@ extern "C" {
  *  major minor fix final patch/beta)
  */
 # define OPENSSL_VERSION_NUMBER  0x1010100fL
-# define OPENSSL_VERSION_TEXT    "OpenSSL 1.1.1  11 Sep 2018"
+# define OPENSSL_VERSION_TEXT    "OpenSSL 1.1.1 FIPS 11 Sep 2018"
 
 /*-
  * The macros below are to be used for shared library (.so, .dll, ...)
