--- openssl-1.0.2a/crypto/bio/bss_dgram.c.orig	2015-03-21 09:18:10.961845296 +0000
+++ openssl-1.0.2a/crypto/bio/bss_dgram.c	2015-03-21 09:18:24.481845528 +0000
@@ -1352,7 +1352,7 @@
                 (socklen_t) (sizeof(sctp_assoc_t) + 256 * sizeof(uint8_t));
             authchunks = OPENSSL_malloc(optlen);
             if (!authchunks) {
-                BIOerr(BIO_F_DGRAM_SCTP_READ, ERR_R_MALLOC_ERROR);
+                BIOerr(BIO_F_DGRAM_SCTP_READ, ERR_R_MALLOC_FAILURE);
                 return -1;
             }
             memset(authchunks, 0, sizeof(optlen));
@@ -1424,7 +1424,7 @@
         char *tmp;
         data->saved_message.bio = b;
         if(!(tmp = OPENSSL_malloc(inl))) {
-            BIOerr(BIO_F_DGRAM_SCTP_WRITE, ERR_R_MALLOC_ERROR);
+            BIOerr(BIO_F_DGRAM_SCTP_WRITE, ERR_R_MALLOC_FAILURE);
             return -1;
         }
         if (data->saved_message.data)
