diff -up openssl-1.1.1d/Configurations/unix-Makefile.tmpl.3~ openssl-1.1.1d/Configurations/unix-Makefile.tmpl
--- openssl-1.1.1d/Configurations/unix-Makefile.tmpl.3~	2020-02-04 14:19:41.050644897 +0100
+++ openssl-1.1.1d/Configurations/unix-Makefile.tmpl	2020-02-04 14:22:19.649651890 +0100
@@ -544,7 +544,7 @@ install_sw: install_dev install_engines
 
 uninstall_sw: uninstall_runtime uninstall_engines uninstall_dev
 
-install_docs: install_man_docs install_html_docs
+install_docs: install_man_docs
 
 uninstall_docs: uninstall_man_docs uninstall_html_docs
 	$(RM) -r $(DESTDIR)$(DOCDIR)
