--- openssl-3.0.0-alpha16/Configurations/10-main.conf.omv	2021-05-06 12:15:03.000000000 +0000
+++ openssl-3.0.0-alpha16/Configurations/10-main.conf	2021-05-19 12:20:08.852035151 +0000
@@ -702,10 +702,20 @@
         ex_libs          => add(threads("-latomic")),
         bn_ops           => "BN_LLONG RC4_CHAR",
     },
+    "linux-generic32-clang" => {
+        inherit_from     => [ "linux-generic32" ],
+        CC               => "clang",
+        CXX              => "clang++",
+    },
     "linux-generic64" => {
         inherit_from     => [ "linux-generic" ],
         bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
     },
+    "linux-generic64-clang" => {
+        inherit_from     => [ "linux-generic64" ],
+        CC               => "clang",
+        CXX              => "clang++",
+    },
 
     "linux-ppc" => {
         inherit_from     => [ "linux-generic32" ],
@@ -767,11 +777,21 @@ my %targets = (
         asm_arch         => 'armv4',
         perlasm_scheme   => "linux32",
     },
+    "linux-armv4-clang" => {
+        inherit_from     => [ "linux-armv4" ],
+        CC               => "clang",
+        CXX              => "clang++",
+    },
     "linux-aarch64" => {
         inherit_from     => [ "linux-generic64" ],
         asm_arch         => 'aarch64',
         perlasm_scheme   => "linux64",
     },
+    "linux-aarch64-clang" => {
+        inherit_from     => [ "linux-aarch64" ],
+        CC               => "clang",
+        CXX              => "clang++",
+    },
     "linux-arm64ilp32" => {  # https://wiki.linaro.org/Platform/arm64-ilp32
         inherit_from     => [ "linux-generic32" ],
         cflags           => add("-mabi=ilp32"),
