--- openssl-1.1.0a/engines/afalg/e_afalg.c.0101~	2016-09-22 12:14:53.000000000 +0200
+++ openssl-1.1.0a/engines/afalg/e_afalg.c	2016-09-22 16:43:01.553835865 +0200
@@ -21,6 +21,7 @@
 
 #include <sys/socket.h>
 #include <linux/version.h>
+#include <sys/eventfd.h>
 #define K_MAJ   4
 #define K_MIN1  1
 #define K_MIN2  0
@@ -102,11 +103,6 @@ static ossl_inline int io_setup(unsigned
     return syscall(__NR_io_setup, n, ctx);
 }
 
-static ossl_inline int eventfd(int n)
-{
-    return syscall(__NR_eventfd, n);
-}
-
 static ossl_inline int io_destroy(aio_context_t ctx)
 {
     return syscall(__NR_io_destroy, ctx);
@@ -152,7 +148,7 @@ static int afalg_setup_async_event_notif
              * waitfd is not set in ASYNC_WAIT_CTX, create a new one
              * and set it. efd will be signaled when AIO operation completes
              */
-            aio->efd = eventfd(0);
+            aio->efd = eventfd(0, 0);
             if (aio->efd == -1) {
                 ALG_PERR("%s: Failed to get eventfd : ", __func__);
                 AFALGerr(AFALG_F_AFALG_SETUP_ASYNC_EVENT_NOTIFICATION,
@@ -176,7 +172,7 @@ static int afalg_setup_async_event_notif
         aio->mode = MODE_ASYNC;
     } else {
         /* Sync mode */
-        aio->efd = eventfd(0);
+        aio->efd = eventfd(0, 0);
         if (aio->efd == -1) {
             ALG_PERR("%s: Failed to get eventfd : ", __func__);
             AFALGerr(AFALG_F_AFALG_SETUP_ASYNC_EVENT_NOTIFICATION,
