--- ssl/ssltest.c	2011-12-14 17:18:03.000000000 -0500
+++ ssl/ssltest.c.oden	2012-03-15 06:11:11.577157097 -0400
@@ -1070,19 +1070,8 @@ bad:
 #ifndef OPENSSL_NO_KRB5
 	if (c_ssl  &&  c_ssl->kssl_ctx)
                 {
-                char	localhost[MAXHOSTNAMELEN+2];
-
-		if (gethostname(localhost, sizeof localhost-1) == 0)
-                        {
-			localhost[sizeof localhost-1]='\0';
-			if(strlen(localhost) == sizeof localhost-1)
-				{
-				BIO_printf(bio_err,"localhost name too long\n");
-				goto end;
-				}
 			kssl_ctx_setstring(c_ssl->kssl_ctx, KSSL_SERVER,
-                                localhost);
-			}
+                                "localhost");
 		}
 #endif    /* OPENSSL_NO_KRB5  */
 
