diff -Naur openssl-3.0.0-beta1/Configurations/10-main.conf openssl-3.0.0-beta1.tpg/Configurations/10-main.conf
--- openssl-3.0.0-beta1/Configurations/10-main.conf	2021-06-17 13:03:42.000000000 +0000
+++ openssl-3.0.0-beta1.tpg/Configurations/10-main.conf	2021-06-19 12:10:17.675525680 +0000
@@ -703,10 +703,20 @@
         inherit_from     => [ "linux-generic32" ],
         ex_libs          => add(threads("-latomic")),
     },
+    "linux-generic32-clang" => {
+        inherit_from     => [ "linux-generic32" ],
+        CC               => "clang",
+        CXX              => "clang++",
+    },
     "linux-generic64" => {
         inherit_from     => [ "linux-generic32" ],
         bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
     },
+    "linux-generic64-clang" => {
+        inherit_from     => [ "linux-generic64" ],
+        CC               => "clang",
+        CXX              => "clang++",
+    },
 
     "linux-ppc" => {
         inherit_from     => [ "linux-latomic" ],
@@ -768,11 +778,21 @@
         asm_arch         => 'armv4',
         perlasm_scheme   => "linux32",
     },
+    "linux-armv4-clang" => {
+        inherit_from     => [ "linux-armv4" ],
+        CC               => "clang",
+        CXX              => "clang++",
+    },
     "linux-aarch64" => {
         inherit_from     => [ "linux-generic64" ],
         asm_arch         => 'aarch64',
         perlasm_scheme   => "linux64",
     },
+    "linux-aarch64-clang" => {
+        inherit_from     => [ "linux-aarch64" ],
+        CC               => "clang",
+        CXX              => "clang++",
+    },
     "linux-arm64ilp32" => {  # https://wiki.linaro.org/Platform/arm64-ilp32
         inherit_from     => [ "linux-generic32" ],
         cflags           => add("-mabi=ilp32"),
